### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'append'

### ISSUE THREAD

**Issue by razby** [lyz-code/mkdocs-newsletter#57]
I got an exception after starting mkdocs with enabled mkdocs-newsletter plugin:

Traceback (most recent call last):
  File "/usr/local/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/mkdocs/__main__.py", line 173, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/mkdocs/commands/serve.py", line 54, in serve
    config = builder()
  File "/usr/local/lib/python3.9/site-packages/mkdocs/commands/serve.py", line 49, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/usr/local/lib/python3.9/site-packages/mkdocs/commands/build.py", line 249, in build
    config = config['plugins'].run_event('config', config)
  File "/usr/local/lib/python3.9/site-packages/mkdocs/plugins.py", line 94, in run_event
    result = method(item, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/mkdocs_newsletter/entrypoints/mkdocs_plugin.py", line 68, in on_config
    config = build_nav(config, newsletter_dir)
  File "/usr/local/lib/python3.9/site-packages/mkdocs_newsletter/services/nav.py", line 68, in build_nav
    return _nav_data_to_nav(nav_data, config)
  File "/usr/local/lib/python3.9/site-packages/mkdocs_newsletter/services/nav.py", line 113, in _nav_data_to_nav
    config["nav"].append({"Newsletters": newsletter_nav})
AttributeError: 'NoneType' object has no attribute 'append'

**Comment by lyz-code** [comment]
Hi @razby, thanks for opening the issue and using mkdocs-newsletter :)

It looks like your `mkdocs.yaml` doesn't have a `nav` section, you can check [the official docs for more information on what that is](https://www.mkdocs.org/user-guide/configuration/#nav), is this intended? or are you just starting your new documentation site, and plan to add it?

**Comment by razby** [comment]
Hi @lyz-code 

It looks like 'mkdocs-newsletter' and 'awesome-pages' plugin not works together. I get this error for mkdocs-newsletter every time when plugin awesome-pages is enabled because of rest filter patterns. :(

nav exsample:
`
nav:
    - ... | index.md
    - ... | regex=Test01/
    - ... | regex=Test02/
`

**Comment by lyz-code** [comment]
We use the `nav` section contents to define the order of the elements of the newsletters, the strong point (as far as I can see) of awesome-pages is that you don't have to define the `nav` section. One plugin needs it, and the other's magic is that you don't use it at all, or use a special syntax that the awesome-plugin interprets to build the nav on the fly. The issue is that it doesn't persist that nav into the `mkdocs.yml` afterwards for other plugins to process it.

Long story short, right now you can't use both plugins together, and I don't think it's trivial to change mkdocs-newsletter to work in this case. Nevertheless if you have a proposal, and the intention to do a PR with it, I'm open to discuss it.

**Comment by lyz-code** [comment]
@razby do you have any further feedback on the issue? or should I close it?

**Comment by razby** [comment]
Hi @lyz-code ,
Thanks for information. The plugin mkdocs-newsletter is a great solution but unfortunately we can't go without awesome-pages plugin.
The ticket can be closed.
